controllers/vspherevm_controller.go

```diff
@@ -351,13 +363,34 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	return reconcile.Result{}, nil
 }
 
-func (r vmReconciler) reconcileNetwork(ctx *context.VMContext, vm infrav1.VirtualMachine) {
+func (r vmReconciler) isWaitingForStaticIPAllocation(ctx *context.VMContext) bool {
+	waitForIP := false
+	devices := ctx.VSphereVM.Spec.Network.Devices
+
+	for _, dev := range devices {
+		if !dev.DHCP4 && !dev.DHCP6 && len(dev.IPAddrs) == 0 {
+			// Static IP is not available yet
+			waitForIP = true
+		}
+	}
+
+	return waitForIP
+}
+
+func (r vmReconciler) reconcileNetwork(ctx *context.VMContext, vm infrav1.VirtualMachine) (bool, error) {
```

Comment:
This function never returns a non-nil error. Please change the signature to only return a boolean.

Reply:
Ack